flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV120.java

```diff
@@ -100,20 +109,17 @@ public ObjectInspector getObjectInspectorForConstant(PrimitiveTypeInfo primitive
 				className = "org.apache.hadoop.hive.serde2.objectinspector.primitive.JavaConstantStringObjectInspector";
 				return HiveReflectionUtils.createConstantObjectInspector(className, value);
 			case CHAR:
-				className = "org.apache.hadoop.hive.serde2.objectinspector.primitive.JavaConstantHiveCharObjectInspector";
-				return HiveReflectionUtils.createConstantObjectInspector(className, value);
+				return new FlinkJavaConstantHiveCharObjectInspector((CharTypeInfo) primitiveTypeInfo, (HiveChar) value);
 			case VARCHAR:
-				className = "org.apache.hadoop.hive.serde2.objectinspector.primitive.JavaConstantHiveVarcharObjectInspector";
-				return HiveReflectionUtils.createConstantObjectInspector(className, value);
+				return new FlinkJavaConstantHiveVarcharObjectInspector((VarcharTypeInfo) primitiveTypeInfo, (HiveVarchar) value);
```

Comment:
Can you explain more why you do this?
What the difference between `JavaConstantHiveVarcharObjectInspector` and `FlinkJavaConstantHiveVarcharObjectInspector`?